pkg/pub_package_reader/lib/pub_package_reader.dart

```diff
@@ -760,3 +760,55 @@ Iterable<ArchiveIssue> checkFunding(String pubspecContent) sync* {
     }
   }
 }
+
+Iterable<ArchiveIssue> checkTopics(String pubspecContent) sync* {
+  final map = loadYaml(pubspecContent) as Map;
+  if (!map.containsKey('topics')) {
+    return;
+  }
+  final topics = map['topics'];
+  if (topics == null || topics is! List || topics.isEmpty) {
+    yield ArchiveIssue(
+        '`pubspec.yaml` has invalid `topics`: only a list of topic names are allowed.');
+    return;
+  }
+  if (topics.length > 5) {
+    yield ArchiveIssue(
+        '`pubspec.yaml` has invalid `topics`: at most 5 topics are allowed.');
+    return;
+  }
+
+  for (var item in topics) {
+    if (item is! String) {
+      yield ArchiveIssue(
+          'Invalid `topics` value (`$item`): only strings are allowed.');
+      continue;
+    }
+    item = item.trim();
+    if (item.length < 3) {
+      yield ArchiveIssue(
+          'Invalid `topics` value (`$item`): name is too short (less than 3 characters).');
+      continue;
+    }
+
+    if (item.length > 32) {
+      yield ArchiveIssue(
+          'Invalid `topics` value (`$item`): name is too long (over 32 characters).');
+      continue;
+    }
+
+    if (topics.where((x) => x == item).length > 1) {
+      yield ArchiveIssue(
+          'Invalid `topics` value (`$item`): name must only be present once.');
```

Comment:
This will be reported twice if we don't `break` here.

Reply:
done.